test/gtest/mhp/communicator.cpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+
+template <typename T> class Communicator : public testing::Test {};
+
+using T = int;
+using DV = dr::mhp::distributed_vector<T>;
+
+TEST(Communicator, Alltoallv) {
+  if (comm_size <= 1)
+    return;
+
+  const std::size_t SIZE = 2;
```

Comment:
Problems with the test
- too much logic in this test (double for loops, while best there should none) 
- it does not test different values being in send and recv displacements vectors (if you accidentaly swap them in the code, the test will not catch it), 
- does not test different sendsizes and recvsizes (if these vectors are mixed in the code it is not caught by test and when values in these vectors are taken from wrong places it is not caught by test)

I propose to 
- create test in 3-ranks only suite.
- have hardcoded all vectors with some different values (including zeros) (like `[1,2,0,4]`)
- at the end test against hardcoded expected value
In this way you will test everything and it will be easy to test reader to verify final value at the end.

Reply:
Skip applying this comment as it is going to be applied separately in https://github.com/oneapi-src/distributed-ranges/issues/412 by @haichangsi